includes/widgets/common.php

```diff
@@ -981,6 +981,498 @@ protected function register_controls() {
 			]
 		);
 
+		$this->start_controls_tabs( '_tabs_positioning' );
+
+		foreach ( [ 'normal', 'hover' ] as $tab ) {
+			$hover_state = 'hover' === $tab ? ':hover' : '';
+
+			$this->start_controls_tab(
+				"_tab_positioning_{$tab}",
+				[
+					'label' => 'normal' === $tab ? __( 'Normal', 'elementor' ) : __( 'Hover', 'elementor' ),
+				]
+			);
+
+			$this->add_control(
+				"_{$tab}_transform_rotate_popover",
+				[
+					'label' => __( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::POPOVER_TOGGLE,
+					'prefix_class' => 'e-',
+					'return_value' => 'transform',
+				]
+			);
+
+			$this->start_popover();
+
+			$this->add_responsive_control(
+				"_{$tab}_transform_rotateZ_effect",
+				[
+					'label' => __( 'Rotate', 'elementor' ),
+					'type' => Controls_Manager::SLIDER,
+					'range' => [
+						'deg' => [
+							'min' => -360,
+							'max' => 360,
+						],
+					],
+					'default' => [
+						'size' => 0,
+					],
+					'selectors' => [
+						"{{WRAPPER}} > .elementor-widget-container{$hover_state}" => '--e-transform-rotateZ: {{SIZE}}deg',
```

Comment:
The value that the variable get is between 0-100 deg, so if you set this control to 100 you get 100deg instead of 360deg.

Reply:
I don't understand, the min-max values are between -360 to 360 and not 0 to 100
![image](https://user-images.githubusercontent.com/79147990/124435132-57493c80-dd7d-11eb-84f2-388fa77b8ec3.png)

Reply:
Following our conversation, please check again.

Reply:
done